src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Pipeline/Steps/ExceptionEventsPipelineStep.cs

```diff
@@ -40,12 +41,64 @@ public void Invoke(Exception exception)
             // the events once the listener is registered to effectively "catch up".
             if (_eventSource.IsEnabled())
             {
-                ulong identifier = _identifierCache.GetOrAdd(new ExceptionIdentifier(exception));
+                ReadOnlySpan<StackFrame> stackFrames = ComputeEffectiveCallStack(exception);
 
-                _eventSource.ExceptionInstance(identifier, exception.Message);
+                ulong identifier = _identifierCache.GetOrAdd(new ExceptionIdentifier(exception, stackFrames));
+
+                ulong[] frameIds = _identifierCache.GetOrAdd(stackFrames);
+
+                _eventSource.ExceptionInstance(
+                    identifier,
+                    exception.Message,
+                    frameIds);
             }
 
             _next(exception);
         }
+
+        private static ReadOnlySpan<StackFrame> ComputeEffectiveCallStack(Exception exception)
```

Comment:
Let's validate this works for reverse p/invoke scenarios.

Reply:
When throwing an exception from a reverse p/invoke (native code calling managed code), the call stack only contains the managed frames; so it works, but there's no indication that manged-to-native or native-to-managed transitions have occurred. We might be able to inspect each method and see if there are a pair of matching extern methods:
<img width="467" alt="reverse" src="https://user-images.githubusercontent.com/17344677/233201491-749aaf72-3bc4-493c-8cef-b2604e11f484.png">
Because the managed callstack is intact, I'll save transition indication for a separate work item.